pkg/adapter/options.go

```diff
@@ -0,0 +1,74 @@
+package adapter
+
+import (
+	"context"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/rollkit/go-execution-abci/modules/network/types"
+)
+
+// Option is a functional option for configuring the Adapter.
+type Option func(*Adapter)
+
+// WithMetrics sets custom metrics for the Adapter.
+func WithMetrics(m *Metrics) Option {
+	return func(a *Adapter) {
+		a.metrics = m
+	}
+}
+
+// WithBlockFilter sets a custom block publisher for the Adapter.
+func WithBlockFilter(publisher BlockFilter) Option {
+	return func(a *Adapter) {
+		a.blockFilter = publisher
+	}
+}
+
+// WithNetworkSoftConfirmationBlockFilter creates a BlockFilter that uses the network module's SoftConfirmationStatus.
+func WithNetworkSoftConfirmationBlockFilter() Option {
+	return func(a *Adapter) {
+		a.blockFilter = &NetworkSoftConfirmationBlockFilter{
+			app: a.App,
+		}
+	}
+}
+
+// NetworkSoftConfirmationBlockFilter is a BlockFilter implementation that uses the network module's SoftConfirmationStatus.
+type NetworkSoftConfirmationBlockFilter struct {
+	app servertypes.ABCI
+}
+
+// IsPublishable implements the BlockFilter interface.
+func (f *NetworkSoftConfirmationBlockFilter) IsPublishable(ctx context.Context, height int64) bool {
```

Comment:
With this option, an abci query is send to the app to check for soft-confirmation by the validator network